### PR TITLE
View Posts

### LINKED ISSUE #45: View All Posts
As a reader, I would like to see a list of all the Posts so that I can choose a post that seems interesting to read.

**Given** the user is in the Tabloid application  
**When** they select the `Posts` menu option  
**Then** they should be directed to the Posts list page  
**And** each post in the list should display the title, author and category  
**And** the list should ONLY contain approved Posts  
**And** the list should ONLY contain Posts with a publication date that is in the past  
**And** the list should be in order of publication date with the most recent on top

### PR DESCRIPTION
**Pull request by cjgochanour**
#### Changes Made
1. Added Post module
2. Added PostList module
3. Added postManager module
4. Added Post model
5. Added Tag model
6. Added Category model
7. Added PostRepository and its interface with GetPublishedPosts method
8. Added Post Controller
9. Added routing to "/posts"
10. Added link to post list in nav bar
​
#### Steps to Review
1. Checkout this branch locally.
    ```
    git fetch --all
    git checkout cg-viewPosts
    ```
2. Open a new Terminal tab and navigate to the server directory.
3. Test app functionality.
     > npm start to run app
     > before logging in, notice that you may not navigate to /posts when unauthorized
     > log in
     > use nav bar link to route to /posts
     > observe list of posts generated
 
4. View code file.
    > Confirm file modifications are present as indicated above.
    > Confirm no unused code or extraneous comments exist.

Closes #45

### CODE REVIEW

File: Tabloid/Models/Post.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Tabloid.Models
+{
+    public class Post
+    {
+        public int Id { get; set; }
```
Comment (joshdbarton) [inline Tabloid/Models/Post.cs:10]: Would it be helpful to add some Data annotations here for model validation?

File: Tabloid/Repositories/PostRepository.cs [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+                    {
+                        Id = reader.GetInt32(reader.GetOrdinal("UserTypeId")),
+                        Name = reader.GetString(reader.GetOrdinal("UserTypeName"))
+                    }
+                },
+                Tags = new List<Tag>()
+            };
+            try
+            {
+                if (!reader.IsDBNull("TagId"))
```
Comment (joshdbarton) [inline Tabloid/Repositories/PostRepository.cs]: Is there a Utils method you could use to do this instead?

File: Tabloid/Repositories/PostRepository.cs [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                    UserTypeId = reader.GetInt32(reader.GetOrdinal("UserTypeId")),
+                    UserType = new UserType()
+                    {
+                        Id = reader.GetInt32(reader.GetOrdinal("UserTypeId")),
+                        Name = reader.GetString(reader.GetOrdinal("UserTypeName"))
+                    }
+                },
+                Tags = new List<Tag>()
+            };
+            try
```
Comment (joshdbarton) [inline Tabloid/Repositories/PostRepository.cs]: what's the try/catch block for?

**Review by joshdbarton — CHANGES_REQUESTED** [review]

**Comment by user1hunter** [comment]
The posts appear when I click the posts menu option. Looks good to me.

File: Tabloid/Repositories/PostRepository.cs (hunk 3 above)
Comment (cjgochanour) [reply Tabloid/Repositories/PostRepository.cs]: Removed try/catch

File: Tabloid/Repositories/PostRepository.cs (hunk 2 above)
Comment (cjgochanour) [reply Tabloid/Repositories/PostRepository.cs]: Now using IsNotDBNull dbutils method

File: Tabloid/Models/Post.cs (hunk 1 above)
Comment (cjgochanour) [reply Tabloid/Models/Post.cs:10]: added data annotations to all new models

**Review by joshdbarton — APPROVED** [review]
:shipit:

**Review by user1hunter — APPROVED** [review]
Looks good.

**Pull request merged**